### ISSUE TITLE
FileNotFoundError: [Errno 2] No such file or directory: '/mnt/models'

### ISSUE THREAD

**Issue by Rstar1998** [kserve/kserve#2986]
I am using kserve version v0.10 and when do kubectl apply -f myyaml.yml then I get the following error. 
Any help on how to solve this issue ? 

YAML :
```
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  name: my-model
spec:
  predictor:
    sklearn:
      protocolVersion: v1
      storageUri: https://github.com/tduffy000/kfserving-uri-examples/blob/master/sklearn/frozen/model.joblib?raw=true
```
Error log : 
```
Defaulted container "kserve-container" out of: kserve-container, queue-proxy
2023-06-13 11:52:08.507 1 root INFO [download():63] Copying contents of /mnt/models to local
2023-06-13 11:52:08.508 1 root ERROR [<module>():39] fail to locate model file for model my-model under dir /mnt/models,trying loading from model repository.
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/sklearnserver/sklearnserver/__main__.py", line 42, in <module>
    kserve.ModelServer(registered_models=SKLearnModelRepository(args.model_dir)).start(
  File "/sklearnserver/sklearnserver/sklearn_model_repository.py", line 24, in __init__
    self.load_models()
  File "/kserve/kserve/model_repository.py", line 37, in load_models
    for name in os.listdir(self.models_dir):
FileNotFoundError: [Errno 2] No such file or directory: '/mnt/models'

```

**Comment by wjp-cn** [comment]
Hello, my storage-initializer container can download this model normally,I guess this is your network issue.But the kserver-container will report an error: "ModuleNotFoundError: No module named 'sklearn. svm. classes'". This is because this Python class has been removed in version 0.24 and you need to find an alternative solution,

**Comment by Manindhar** [comment]
My issue is also similar - https://github.com/kserve/kserve/issues/3082

**Comment by sivanantha321** [comment]
This maybe releated to this issue https://github.com/kserve/kserve/issues/1462 and it is fixed by #1463. It should be working now.

**Comment by amacharya** [comment]
Hello,

kserve v0.11.0
kind - kubernetes v1.25

I am still facing an issue.  - FileNotFoundError: [Errno 2] No such file or directory: '/mnt/models'

any solution/workaround?
